src/svg-renderer.js

```diff
@@ -316,6 +350,8 @@ export default class SVGRenderer {
    */
   renderNodesDelta() {
     const chart = this.chart;
+    const oldNodeMap = this.oldNodeMap;
+    const useStableLayout = this.options.useStableLayout && (flatten(this.layout).nodes.length - 1) < chart.selectAll('.node').size();
```

Comment:
I'm not sure I follow why each of these conditions checks the new vs. existing length, and especially why this one subtracts 1 first. Can you give me a brief explanation?

Reply:
Kind of a quick and dirty hack - the nodes have an extra parent-container so the numbers of visible nodes are off by 1.

As for the length, I want to say if you are adding things then I can't make stableness guarantees, so to do that we do a simple check of number of nodes/edges now versus previous counts. This doesn't work all the time, as per description it does not take hierarchy into account.

Reply:
Ah okay. Might be worth adding a comment explaining that for future readers, your call.
Thanks!